mllib/src/main/scala/org/apache/spark/ml/feature/StopWordsRemover.scala

```diff
@@ -84,7 +86,36 @@ class StopWordsRemover @Since("1.5.0") (@Since("1.5.0") override val uid: String
   @Since("1.5.0")
   def getCaseSensitive: Boolean = $(caseSensitive)
 
-  setDefault(stopWords -> StopWordsRemover.loadDefaultStopWords("english"), caseSensitive -> false)
+  /**
+   * [[https://docs.oracle.com/javase/8/docs/api/java/util/Locale.html Locale]] of the input.
+   * Ignored when [[caseSensitive]] is false.
+   * Default: Locale.English
+   * @see `StopWordsRemover.loadDefaultStopWords()`
+   * @group param
+   */
+  @Since("2.4.0")
+  val locale: Param[Locale] = new Param[Locale](this, "locale",
+    "Locale of the input. Ignored when caseSensitive is false.")
+
+  /**
+   * @group setParam
+   * @throws IllegalArgumentException if `value` is not an available locale.
+   */
+  @Since("2.4.0")
+  def setLocale(value: Locale): this.type = {
+    if (Locale.getAvailableLocales contains value) {
```

Comment:
Could you add the check as a param validator? See https://github.com/dongjinleekr/spark/blob/master/mllib/src/main/scala/org/apache/spark/ml/param/params.scala#L167. This is because there are more than one way to set params besides this setter.